### ISSUE TITLE
Builds finished but no artifacts were created.please help

### ISSUE THREAD

**Issue by kedllpbs** [rapid7/metasploitable3#436]
After a few attempts and searches I haven't been able to solve the problem I tried to create metasploitable3.
## Host System
* OS:win10
* Packer Version:Packer v1.5.1
* Vagrant Version:Vagrant 2.2.6
* VirtualBox Version:6.0

Compatible version of VirtualBox found.
Compatible version of Packer found.
Compatible version of Vagrant found.
Compatible version of vagrant-reload plugin found.
All requirements found. Proceeding...
Building metasploitable3-win2k8 Vagrant box...
virtualbox-iso: output will be in this color.

==> virtualbox-iso: Retrieving Guest additions
==> virtualbox-iso: Trying C:\Program Files\Oracle\VirtualBox/VBoxGuestAdditions.iso
==> virtualbox-iso: Trying file://C:/Program%20Files/Oracle/VirtualBox/VBoxGuestAdditions.iso

==> Builds finished but no artifacts were created.
Error building the Vagrant box using Packer. Please check the output above for any error messages.
At C:\Users\User\Desktop\Metasploitable3\metasploitable3\build.ps1:157 char:10
+          throw "Error building the Vagrant box using Packer. Please c ...
+          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (Error building ...error messages.:String) [], RuntimeException
    + FullyQualifiedErrorId : Error building the Vagrant box using Packer. Please check the output above for any error messages.

**Comment by drknite72** [comment]
I ran in to the same problem you did on the window 2008 build.
What i found in the logs it was failing on elasticsearch install. Issue #435 mentioned http was not working but https was. My guess it was updated for Ubuntu but not 2k8, have not run the Ubunutu build yet so dont know.
anyway, located install_elasticsearch.bat under /scripts/installs and the download file is http, copied the url to a browser and got the 501 error, changed it to https and got the download.
running the script again now to see if it fixed my issue.
Will update if it does.

**Comment by drknite72** [comment]
So i was able to build the windows VM after editing the elasticsearch bat file to https for the download. I also needed to wipe out the cached files and partially created vm's before it want through

**Comment by filippossemere** [comment]
> So i was able to build the windows VM after editing the elasticsearch bat file to https for the download. I also needed to wipe out the cached files and partially created vm's before it want through

Where can I find this elasticsearch bat file and the cached files which needs to be removed?

**Comment by jayimu** [comment]
Sorry, I didn't encounter the above problem.


| |
J
|
|
nf.s@163.com
|
签名由网易邮箱大师定制


On 02/6/2020 17:06，oogabooga<notifications@github.com> wrote：

So i was able to build the windows VM after editing the elasticsearch bat file to https for the download. I also needed to wipe out the cached files and partially created vm's before it want through

Where can I find this elasticsearch bat file and the cached files which needs to be removed?

**Comment by filippossemere** [comment]
OK, I suspect I got the error in the pastebin above because of Hyper-V. I just ran the same script on another laptop and it worked perfectly after I activated VT-X.

**Comment by jayimu** [comment]
sure，I use vmware, so I have n’t encountered problems like yours。


| |
J
|
|
nf.s@163.com
|
签名由网易邮箱大师定制


On 02/6/2020 22:57，oogabooga<notifications@github.com> wrote：

OK, I suspect I got the error in the pastebin above because of Hyper-V. I just ran the same script on another laptop and it worked perfectly after I activated VT-X.

**Comment by filippossemere** [comment]
I lied, the VM successfully rebooted multiple times and installed ssh and other features but stopped on something called 'chocolately'.

 ```
   virtualbox-iso: Enabling feature(s)
    virtualbox-iso: The operation completed successfully.
==> virtualbox-iso: Provisioning with powershell script: C:\Users\fs-test\Downloads\metasploitable3-master\packer\templates/../../scripts/installs/setup_snmp.bat
    virtualbox-iso:
    virtualbox-iso: Deployment Image Servicing and Management tool
    virtualbox-iso: Version: 6.1.7600.16385
    virtualbox-iso:
    virtualbox-iso: Image Version: 6.1.7600.16385
    virtualbox-iso:
    virtualbox-iso: Enabling feature(s)
    virtualbox-iso: The operation completed successfully.
    virtualbox-iso: The operation completed successfully.
    virtualbox-iso:
    virtualbox-iso: The operation completed successfully.
    virtualbox-iso:
    virtualbox-iso: The operation completed successfully.
    virtualbox-iso:
==> virtualbox-iso: Provisioning with windows-shell...
==> virtualbox-iso: Provisioning with shell script: C:\Users\fs-test\Downloads\metasploitable3-master\packer\templates/../../scripts/installs/install_boxstarter.bat
==> virtualbox-iso: 'chocolatey' is not recognized as an internal or external command,
==> virtualbox-iso: operable program or batch file.
    virtualbox-iso:
    virtualbox-iso: C:\Users\vagrant>chocolatey feature enable -n=allowGlobalConfirmation
    virtualbox-iso:
    virtualbox-iso: C:\Users\vagrant>choco install BoxStarter
    virtualbox-iso:
    virtualbox-iso: C:\Users\vagrant>chocolatey feature disable -n=allowGlobalConfirmation
==> virtualbox-iso: 'choco' is not recognized as an internal or external command,
==> virtualbox-iso: operable program or batch file.
==> virtualbox-iso: 'chocolatey' is not recognized as an internal or external command,
==> virtualbox-iso: operable program or batch file.
==> virtualbox-iso: Provisioning step had errors: Running the cleanup provisioner, if present...
==> virtualbox-iso: Deregistering and deleting VM...
==> virtualbox-iso: Deleting output directory...
Build 'virtualbox-iso' errored: Script exited with non-zero exit status: 1.Allowed exit codes are: [0]

==> Some builds didn't complete successfully and had errors:
--> virtualbox-iso: Script exited with non-zero exit status: 1.Allowed exit codes are: [0]

==> Builds finished but no artifacts were created.
Error building the Vagrant box using Packer. Please check the output above for any error messages.
At C:\Users\fs-test\Downloads\metasploitable3-master\build.ps1:157 char:10
+          throw "Error building the Vagrant box using Packer. Please c ...
+          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (Error building ...error messages.:String) [], RuntimeException
    + FullyQualifiedErrorId : Error building the Vagrant box using Packer. Please check the output above for any error
    messages.
```

**Comment by checkeralex** [comment]
We have the same problem @filippossemere

**Comment by jmartin-r7** [comment]
Fixed by #439
